commons/zenoh-protocol/src/network/declare.rs

```diff
@@ -668,60 +726,66 @@ pub mod interest {
     ///
-    /// E.g., the [`DeclareInterest`]/[`FinalInterest`]/[`UndeclareInterest`] message flow is the following:
+    /// The behaviour of a DECLARE INTEREST depends on the DECLARE MODE in the DECLARE MESSAGE:
+    /// - Push: only future declarations
+    /// - Request: only current declarations
+    /// - RequestContinous: current and future declarations
+    /// - Response: invalid
+    ///
+    /// E.g., the [`DeclareInterest`] message flow is the following:
     ///
     /// ```text
     ///     A                   B
     ///     |   DECL INTEREST   |
-    ///     |------------------>| -- This is a DeclareInterest e.g. for subscriber declarations/undeclarations.
+    ///     |------------------>| -- Sent in Declare::RequestContinuous.
+    ///     |                   |    This is a DeclareInterest e.g. for subscriber declarations/undeclarations.
     ///     |                   |
     ///     |  DECL SUBSCRIBER  |
-    ///     |<------------------|
+    ///     |<------------------| -- Sent in Declare::Response
     ///     |  DECL SUBSCRIBER  |
-    ///     |<------------------|
+    ///     |<------------------| -- Sent in Declare::Response
     ///     |  DECL SUBSCRIBER  |
-    ///     |<------------------|
+    ///     |<------------------| -- Sent in Declare::Response
     ///     |                   |
-    ///     |   FINAL INTEREST  |
-    ///     |<------------------|  -- The FinalInterest signals that all known subscribers have been transmitted.
+    ///     |       FINAL       |
+    ///     |<------------------| -- Sent in Declare::Response
     ///     |                   |
     ///     |  DECL SUBSCRIBER  |
-    ///     |<------------------|  -- This is a new subscriber declaration.
+    ///     |<------------------| -- Sent in Declare::Push. This is a new subscriber declaration.
     ///     | UNDECL SUBSCRIBER |
-    ///     |<------------------|  -- This is a new subscriber undeclaration.
+    ///     |<------------------| -- Sent in Declare::Push. This is a new subscriber undeclaration.
     ///     |                   |
     ///     |        ...        |
     ///     |                   |
-    ///     |  UNDECL INTEREST  |
-    ///     |------------------>|  -- This is an UndeclareInterest to stop receiving subscriber declarations/undeclarations.
+    ///     |       FINAL       |
+    ///     |------------------>| -- Sent in Declare::RequestContinuous.
+    ///     |                   |    This stops the transmission of subscriber declarations/undeclarations.
     ///     |                   |
     /// ```
     ///
     /// The DECLARE INTEREST message structure is defined as follows:
     ///
     /// ```text
     /// Flags:
-    /// - C: Current        if C==1 then the interest refers to the current declarations.
+    /// - X: Reserved
     /// - F: Future         if F==1 then the interest refers to the future declarations. Note that if F==0 then:
```

Comment:
What's the difference between setting `F==1` on `DeclareInterest` and setting `Mode==Push` on `Declare`?

Reply:
(Meant to select the next line, can't edit it)

Reply:
That's a good point.

DeclareInterest has 3 combinations:
1. Only current
2. Only future
3. Current and future

These could be achieved via:
1. Declare(Request)::Interest -> Only current
2. Declare(Push)::Interest -> Only future
3. Declare(RequestContinous) -> Current and future

If that's the case, we might not even need the `F` flag for future and it can be removed. 
Chiming in @OlivierHecart. What do you think?

Reply:
Indeed we should not need Current neither Future flags any more.

Reply:
Ok, flag removed.